src/image_provider/s3.rs

```diff
@@ -145,27 +146,49 @@ pub mod s3 {
                     })
                     .collect(),
             };
-            let mut binary_payload: Vec<u8> = Vec::new();
-            while let Some(bytes) = result.body.try_next().await.map_err(|e| {
-                error!(
-                    "failed to read the response for the file '{}'. error: '{}'",
-                    resource, e
-                );
-                ImageDownloadFailed
-            })? {
-                binary_payload.write_all(&bytes).map_err(|e| {
+            if let Some(max_size) = config.max_file_size {
```

Comment:
Same remark regarding backward compatibility. Please refactor it to continue downloading the entire file in case the configuration parameter is missing.